src/coreclr/jit/valuenum.cpp

```diff
@@ -6812,6 +6812,47 @@ ValueNum EvaluateBinarySimd(ValueNumStore* vns,
         }
 #endif // TARGET_XARCH
 
+        case TYP_BOOL:
+        {
+            assert((oper == GT_EQ) || (oper == GT_NE));
+
+            var_types vn1Type = vns->TypeOfVN(arg0VN);
+            var_types vn2Type = vns->TypeOfVN(arg1VN);
+
+            assert((vn1Type == vn2Type) && varTypeIsSIMD(vn1Type));
+            assert(!varTypeIsFloating(baseType));
+
+            ValueNum packed     = EvaluateBinarySimd(vns, GT_EQ, scalar, vn1Type, baseType, arg0VN, arg1VN);
```

Comment:
shouldn't this be `EvaluateBinarySimd(vns, oper, ...)`?

Reply:
I wonder if it would be simpler to implement this as `EvaluateVector` and then simplify check result `IsAllBitsSet` or `!IsZero`

Which would also simplify the other relational comparisons

Reply:
> shouldn't this be EvaluateBinarySimd(vns, oper, ...)?

It makes it harder to check output, doesn't it?
Currently I just pass EQ and then return AllBitsSet or !AllBitsSet depending on source oper

Reply:
Does it? We have a simple property for any simd_T and it makes it easier to cover the other cases.

Reply:
That is, if you just evaluate per element you get a result simd_T result and can then just check IsAllBitsSet or IsZero